### PR TITLE
[MU4] Elements select

### PR DESCRIPTION
**Pull request by Eism**
- Transferred select note dialog to MU4
- Transferred select element dialog to MU4

Peek 2020-11-07 16-06

### CODE REVIEW

File: mu4/notation/inotationelements.h [hunk 1]
```diff
@@ -18,6 +19,7 @@ class INotationElements
     virtual ~INotationElements() = default;
 
     virtual Ms::Element* search(const std::string& searchCommand) const = 0;
```
Comment (vpereverzev) [inline mu4/notation/inotationelements.h:21]: From that point it's not really obvious which "select" should be used. Is there any chance to merge these two methods and leave a single way to search?

**Comment by igorkorsukov** [comment]
needs rebase

File: mu4/notation/inotationelements.h (hunk 1 above)
Comment (RomanPudashkin) [reply mu4/notation/inotationelements.h:21]: I think, better to rename this method to something like "elements" and also rename SearchElementOptions to FilterElementsOptions. By default, this method should return all score elements

Comment (RomanPudashkin) [inline mu4/notation/inotationelements.h:21]: Please, rename searchCommand to searchText

File: mu4/notation/inotationinteraction.h [hunk 2]
```diff
@@ -53,7 +53,8 @@ class INotationInteraction
     // Select
     virtual Element* hitElement(const QPointF& pos, float width) const = 0;
     virtual int hitStaffIndex(const QPointF& pos) const = 0;
-    virtual void select(Element* element, SelectType type, int staffIndex = 0) = 0;
+    virtual void select(Element* e, SelectType type, int staffIdx = 0) = 0;
+    virtual void select(std::vector<Element*> elements, SelectType type, int staffIndex = 0) = 0;
```
Comment (RomanPudashkin) [inline mu4/notation/inotationinteraction.h:57]: Better to unite select(Element* e, ...) and select(std::vector<Element*> elements, ...)

File: mu4/notation/inotationelements.h (hunk 1 above)
Comment (Eism) [reply mu4/notation/inotationelements.h]: fixed. Thanks!

**Review by RomanPudashkin — APPROVED** [review]

File: mu4/notation/internal/notationactioncontroller.cpp [hunk 3]
```diff
@@ -478,6 +570,23 @@ void NotationActionController::openPartsDialog()
     interactive()->open("musescore://notation/parts");
 }
 
+FilterElementsOptions NotationActionController::elementsFilterOptions(const Element* element) const
+{
+    FilterElementsOptions options;
+    options.elementType = element->type();
+
+    if (element->type() == ElementType::NOTE) {
+        const Ms::Note* note = dynamic_cast<const Ms::Note*>(element);
```
Comment (igorkorsukov) [inline mu4/notation/internal/notationactioncontroller.cpp:579]: Better use method `toNote`, it have extra checks

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**